src/ui/Program.cs

```diff
@@ -1,19 +1,57 @@
-﻿using Microsoft.AspNetCore;
+﻿using System.Net;
+using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
 
-namespace GovUk.Education.ManageCourses.Ui
-{
-    public class Program
-    {
-        public static void Main(string[] args)
-        {
-            BuildWebHost(args).Run();
+namespace GovUk.Education.ManageCourses.Ui {
+    public class Program {
+        public static void Main (string[] args) {
+            BuildWebHost (args).Run ();
         }
 
-        public static IWebHost BuildWebHost(string[] args) =>
-            WebHost.CreateDefaultBuilder(args)
-                .UseKestrel(options => options.AddServerHeader = false)
-                .UseStartup<Startup>()
-                .Build();
+        private const int DefaultPort = 44364;
+        public static int GetPort (string environment) {
+
+            var port = DefaultPort;
+
+            var envPort = System.Environment.GetEnvironmentVariable ($"MANAGE_COURSES_API_{environment}_PORT");
+            int portNumber;
+            if (int.TryParse (envPort, out portNumber) &&
+                portNumber >= 49152 &&
+                portNumber <= 65535) {
+                port = portNumber;
+            }
+
+            return port;
+        }
+
+        public static IWebHost BuildWebHost (string[] args) {
+
+            return WebHost.CreateDefaultBuilder (args)
+                .UseKestrel (options => {
+                    options.AddServerHeader = false;
+                    var environment = System.Environment.GetEnvironmentVariable ("ASPNETCORE_ENVIRONMENT");
+                    if (environment == "Development") {
+                        var port = GetPort (environment);
```

Comment:
@the-daniel-rothig 

I have refactored the code to be more hygienic especially with the ports as not setting or setting invalid values causes random port assignments. 

I have also expand/amended the readme as well